x-pack/plugins/transform/public/app/common/validators/integer_above_zero_validator.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { numberValidator } from '@kbn/ml-agg-utils';
+
+import type { Validator } from './types';
+
+const numberAboveZeroNotValidErrorMessage = i18n.translate(
+  'xpack.transform.transformList.editFlyoutFormNumberAboveZeroNotValidErrorMessage',
+  {
+    defaultMessage: 'Value needs to be an integer above zero.',
+  }
+);
+
+export const integerAboveZeroValidator: Validator = (value) =>
+  !(value + '').includes('.') && numberValidator({ min: 1, integerOnly: true })(+value) === null
```

Comment:
Can we use a generic validator and avoid creating a function for each specific case?

Reply:
I'm not sure I fully get the suggestion. This is making use of the generic validator `numberValidator`, it then sets this up as a wrapper for a specific use case required by the edit transform flyout to satisfy the spec of the request object to update a transform. Also note this isn't new code, all the validators have just been moved to individual files and a corresponding jest test file, the intent of this PR wasn't to change any logic but just to spread out the existing validators into separate files. Can you share more details about how you envision that generic validator?

Reply:
Why do you need the `!(value + '').includes('.')` part? The generic validator already checks for integers with `ingegerOnly`.

Reply:
We have a test assertion in there that would fail if remove that part: `expect(integerAboveZeroValidator('1.')).toEqual(['Value needs to be an integer above zero.']);`. I guess it's because `1.` technically passes `Number.isInteger` but we didn't want users to be able to enter `1.`.